internals-js/src/schemaUpgrader.ts

```diff
@@ -714,7 +714,7 @@ class SchemaUpgrader {
     // - to every "value type" (in the fed1 sense of non-root type and non-entity) if it is used in any other subgraphs
     // - to any (non-external) field of an entity/root-type that is not a key field and if another subgraphs resolve it (fully or partially through @provides)
     for (const type of this.schema.objectTypes()) {
-      if (type.hasAppliedDirective(keyDirective) || type.isRootType()) {
+      if (type.hasAppliedDirective(keyDirective) || (type.isRootType() && !type.isSubscriptionRootType())) {
```

Comment:
More a general question here, can @shareable be added on fields of Query or Mutation types? If yes:
- what does that mean in terms of planning? When will one subgraph be hit over another?
- the @shareable [spec/doc](https://www.apollographql.com/docs/federation/federated-schemas/sharing-types/#sharing-object-types) only specifies "value types", maybe it would be good to add Query and Mutation types to be included in that case or Subscription to be mentioned.

If not, then we could just by pass root types altogether? i.e
```
if (type.isRooType()) {
  continue;
}
if (type.hasAppliedDirective(keyDirective) {
  ...
} else {
  ...
}
```

Reply:
You are correct.